--- a/avidemux_core/ADM_coreScriptPython/src/CMakeLists.txt	2012-03-18 00:39:10.000000000 -0700
+++ b/avidemux_core/ADM_coreScriptPython/src/CMakeLists.txt	2012-03-18 00:58:09.000000000 -0700
@@ -15,9 +15,11 @@
 include_directories(../../ADM_coreScript/include)
 include_directories(${AVIDEMUX_TOP_SOURCE_DIR}/avidemux/common)
 include_directories(${AVIDEMUX_TOP_SOURCE_DIR}/avidemux/common/ADM_editor/include)
-
-if (APPLE)
+
+target_link_libraries(ADM_coreScriptPython ADM_coreScript ADM_coreUtils6)
+
+if (NOT APPLE)
+	target_link_libraries(ADM_coreScriptPython -Wl,--version-script,"${CMAKE_CURRENT_SOURCE_DIR}/ADM_coreScriptPython.v")
+elseif (NOT ADM_CPU_X86_64)
 	target_link_libraries(ADM_coreScriptPython -Wl,-read_only_relocs,suppress)
-endif (APPLE)
-
-target_link_libraries(ADM_coreScriptPython ADM_coreScript ADM_coreUtils6 -Wl,--version-script,"${CMAKE_CURRENT_SOURCE_DIR}/ADM_coreScriptPython.v")
+endif (NOT APPLE)
